### PR TITLE
use the `rename_prefix_namespace` flag instead of setting it in java

### PR DESCRIPTION
**Pull request by erwinmombay**
to try and remove as much custom settings in the java code, set `rename_prefix_namespace` through the config flag instead.

### CODE REVIEW

File: build-system/compile/single-pass.js [hunk 1]
```diff
@@ -103,6 +103,10 @@ exports.getFlags = function(config) {
     source_map_location_mapping: ['|/'],
     //new_type_inf: true,
     language_in: 'ES6',
+    // This is important for exports across js modules. Because of the async
+    // loading nature of amp scripts we export everything through the `_`
+    // namespace to be references acruss `chunks`.
```
Comment (jridgewell) [inline build-system/compile/single-pass.js]: s/acruss/across/

File: build-system/compile/single-pass.js [hunk 2]
```diff
@@ -103,6 +103,10 @@ exports.getFlags = function(config) {
     source_map_location_mapping: ['|/'],
     //new_type_inf: true,
     language_in: 'ES6',
+    // This is important for exports across js modules. Because of the async
+    // loading nature of amp scripts we export everything through the `_`
```
Comment (jridgewell) [inline build-system/compile/single-pass.js]: Nit: mention wrapped modules preventing export of local variables.

File: build-system/compile/single-pass.js (hunk 1 above)
Comment (erwinmombay) [reply build-system/compile/single-pass.js]: fixed

**Comment by rsimha** [comment]
Nice! This makes https://github.com/ampproject/amphtml/pull/22056/commits/63c7fbcdfbc5400573f64d348fcf00a57d812864 unnecessary.

**Pull request merged**